docs/other_resources/in-process-providers/evaluators/fractional_evaluation.md

```diff
@@ -0,0 +1,211 @@
+# Fractional Evaluation
+
+This evaluator allows to split the returned variants of a feature flag into different buckets,
+where each bucket can be assigned a percentage, representing how many requests will resolve to the corresponding
+variant. The sum of all weights must be 100, and the distribution must be performed by using the value of a referenced
+from the evaluation context to hash that value and map it to a value between [0, 100]. It is important to note
+that evaluations MUST be sticky, meaning that flag resolution requests containing the same value for the
+referenced property in their context MUST always resolve to the same variant. For calculating the hash value of the
+referenced evaluation context property, the [MurmurHash3](https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp)
+hash function should be used. This is to ensure that flag resolution requests yield the same result,
+regardless of which implementation of the in-process flagd provider is being used.
+
+The implementation of this evaluator should accept the object containing the `fractionalEvaluation` evaluator
+configuration, and a `data` object containing the evaluation context. The evaluator configuration should be an
+array containing at least two items, with the first item being a `string`
+value representing the target property to base the distribution of values on, and the remaining items
+being `arrays` with two values, with the first being `string` item representing the name of the variant, and the
+second being a `float` item representing the percentage for that variant. The percentages of all items must add up to
+100.0, otherwise unexpected behavior can occur during the evaluation. The `data` object can be an arbitrary
+JSON object. Below is an example for a targetingRule containing a `fractionalEvaluation`:
+
+```json
+{
+  "flags": {
+    "headerColor": {
+      "variants": {
+        "red": "#FF0000",
+        "blue": "#0000FF",
+        "green": "#00FF00"
+      },
+      "defaultVariant": "red",
+      "state": "ENABLED",
+      "targeting": {
+        "fractionalEvaluation": [
+          "email",
```

Comment:
There's been discussion around this which could make the implementation powerful and still simple. One of the limitations it would fix is that it would allow use to use nested properties for the bucket value... The meat of the change is to use a JSON logic expression to get the bucketing value. This allows for maximum flexibility and _should_ be simple to implement, I believe.

So here, instead of `"email"`, we'd have `{"var":"email"}` or whatever other expression was wanted. It was also suggested we could support leaving this param out completely, in which case, `{"var":"targetingKey"}` would be used.

See [here](https://github.com/open-feature/flagd/pull/847#discussion_r1300362966).

cc @craigpastro

Reply:
@toddbaert I like this suggestion, as it will increase flexibilit. Also, implementation wise it should actually make things easier as the implementer of this logic will not have to take care of retrieving the value of the referenced property from the evaluation context object. Instead, we will have the resolved value already available in the evaluation method (similar to the recently introduced semver/starts_with/ends_with evaluators)

Reply:
It seems like you missed a spot here: https://github.com/open-feature/flagd/pull/848#discussion_r1302996057

Reply:
good catch and thank you for the suggestion - that's fixed now